cirq-google/cirq_google/named_topologies.py

```diff
@@ -0,0 +1,259 @@
+        return nx.from_edgelist(
+            [(i1, i2) for i1, i2 in zip(range(self.n_nodes), range(1, self.n_nodes))]
+        )
+
+    def draw(self, ax=None, cartesian: bool = True, **kwargs) -> Dict[Any, Tuple[int, int]]:
+        """Draw this graph.
+
+        Args:
+            ax: Optional matplotlib axis to use for drawing.
+            cartesian: If True, draw as a horizontal line. Otherwise, draw on a diagonal.
+            kwargs: Additional arguments to pass to `nx.draw_networkx`.
+        """
+        g2 = nx.relabel_nodes(self.graph, {n: (n, 1) for n in self.graph.nodes})
+        return draw_gridlike(g2, ax=ax, cartesian=cartesian, **kwargs)
+
+    def _json_dict_(self):
+        return dataclass_json_dict(self, namespace='cirq.google')
+
+
+@dataclass(frozen=True)
+class DiagonalRectangleTopology(NamedTopology):
+    width: int
+    height: int
+
+    def __post_init__(self):
+        if self.width <= 0:
+            raise ValueError("Width must be a positive integer")
+        if self.height <= 0:
+            raise ValueError("Height must be a positive integer")
+
+    @property  # type: ignore
+    @cache
+    def name(self) -> str:
+        return f'{self.width}-{self.height}-diagonal-rectangle'
+
+    @property  # type: ignore
+    @cache
+    def graph(self) -> nx.Graph:
+        g = nx.Graph()
+        # construct a "diagonal rectangle graph" whose width and height
+        # set the number of rows of 'central' qubits, each of which has
+        # four neighbors in each cardinal direction
+        for megarow in range(self.height):
+            for megacol in range(self.width):
+                y = megacol + megarow
+                x = megacol - megarow
+                g.add_edge((x, y), (x - 1, y))
+                g.add_edge((x, y), (x, y - 1))
+                g.add_edge((x, y), (x + 1, y))
+                g.add_edge((x, y), (x, y + 1))
+        return g
+
+    @property  # type: ignore
+    @cache
+    def n_nodes(self) -> int:
+        return 2 * self.width * self.height + self.width + self.height + 1
+
+    def draw(self, ax=None, cartesian=True, **kwargs):
+        """Draw this graph
+
+        Args:
+            ax: Optional matplotlib axis to use for drawing.
+            cartesian: If True, directly position as (row, column); otherwise,
+                rotate 45 degrees to accommodate the diagonal nature of this topology.
+            kwargs: Additional arguments to pass to `nx.draw_networkx`.
+        """
+        return draw_gridlike(self.graph, ax=ax, cartesian=cartesian, **kwargs)
+
+    def nodes_as_gridqubits(self) -> List['cirq.GridQubit']:
+        """Get the graph nodes as cirq.GridQubit"""
+        import cirq
+
+        return [cirq.GridQubit(r, c) for r, c in sorted(self.graph.nodes)]
+
+    def _json_dict_(self):
+        return dataclass_json_dict(self, namespace='cirq.google')
+
+
+def get_placements(
+    big_graph: nx.Graph, small_graph: nx.Graph, max_placements=100_000
+) -> List[Dict]:
```

Comment:
Missing docstring.  I totally cannot tell what this function is doing from a casual inspection.

Reply:
added